### PR TITLE
Teraterm: Add version 4.103

### LINKED ISSUE #2416: [Request] Add Tera Term
Tera Term is a terminal emulator used for ssh/telnet/serial connection. Useful for Windows users. URL: https://ttssh2.osdn.jp/index.html.en or https://osdn.net/projects/ttssh2/

### PR DESCRIPTION
**Pull request by issaclin32**
close #2416.

**Tera Term** is a terminal emulator for ssh/telnet/serial connection.
Homepage: https://osdn.net/projects/ttssh2/

Please let me know if there are any problems. Thanks.

### CODE REVIEW

**Comment by linsui** [comment]
You can use this link https://osdn.net/dl/ttssh2/teraterm-4.102.zip for url and hash

File: bucket/teraterm.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            "Tera Term"
+        ]
+    ],
+    "checkver": "releases/(?<tag>\\d+)\">Tera Term ([\\d.]+)</a>",
+    "autoupdate": {
+        "url": "http://pumath.dl.osdn.jp/ttssh2/$matchTag/teraterm-$version.zip",
+        "extract_dir": "teraterm-$version",
+        "hash": {
+            "url": "https://osdn.net/projects/ttssh2/downloads/$matchTag/teraterm-$version.zip/",
+            "regex": "SHA256</dt>[\\n\\s]+<dd>([0-9a-f]+)</dd>"
```
Comment (Ash258) [inline bucket/teraterm.json]: Test your autoudpates before posting PR. This obviously do not work without mulitline

```suggestion
            "regex": "(?sm)MD5</dt>[\\n\\s]+<dd>$md5</dd>"
```

File: bucket/teraterm.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "homepage": "https://osdn.net/projects/ttssh2/",
+    "description": "Terminal emulator for ssh/telnet/serial connection.",
+    "version": "4.103",
+    "license": "BSD-3-Clause",
+    "url": "http://pumath.dl.osdn.jp/ttssh2/71232/teraterm-4.103.zip",
+    "hash": "99c9b7559c0551412ed75673f2449d84f25a945e351c7a76689f5751628d56d5",
```
Comment (Ash258) [inline bucket/teraterm.json]: Run autoupdate

File: bucket/teraterm.json (hunk 1 above)
Comment (linsui) [reply bucket/teraterm.json]: It works well for me... Why not use sha256? It's safer.

```
Searching hash for teraterm-4.103.zip in https://osdn.net/dl/ttssh2/teraterm-4.103.zip
Found: 99c9b7559c0551412ed75673f2449d84f25a945e351c7a76689f5751628d56d5 using Extract Mode
```

Comment (issaclin32) [reply bucket/teraterm.json]: Thanks for the suggestion.

I did tested the *autoupdate*. (and it's working, is `multiline` enabled by default?)
hash

**Pull request merged**